upgrade-prechecks/preupgrade-healthcheck.sh

```diff
@@ -948,44 +955,67 @@ function verify_CSI_configmap_present(){
+        else
+            print error "${CHECK_FAIL} token secret is not present in service account for scale"
+        fi
+    else
+        print info "Skipping this test as its is not a metrodr setup having OCP version 4.12"
+    fi
+}
+
 rm -f ${REPORT} > /dev/null
-print_header
-print_section "API access"
-verify_api_access
-print_section "Registry access"
-areRegistriesAccessible
-print_section "Registry authentication and authorisation"
-isAuthCorrect
-print_section "Cluster operators"
-verify_clusteroperators_status
-print_section "Nodes"
-verify_nodes_status
-print_section "Machine configuration pools"
-verify_mcp
-print_section "Catalog sources"
-verify_catsrc
-print_section "Fusion software"
-verify_fusion_health
-print_section "Backup & Restore"
-verify_br_health
-print_section "Legacy SPP"
-verify_spp_health
-print_section "Data Classification"
-verify_dcs_health
-print_section "Scale cluster"
-verify_scale_health
-print_section "VMs migration"
-verify_livemigratable_vms
-print_section "Pods with imagepullbackoff across cluster"
-verify_imagepullbackoff_pods
-print_section "Nodes hardware status"
-verify_nodes_hw
-print_section "Fusion nodes maintenance"
-verify_node_taints
-print_section "Network checks"
-verify_network_checks
-print_section "Verify PID limit"
-verify_pid_limit
-print_section "Verify Presence of CSI Configmap"
-verify_CSI_configmap_present
-print_footer
+if [ "$1" != "postcheck" ]; then
+        print_header
+        print_section "API access"
+        verify_api_access
+        print_section "Registry access"
+        areRegistriesAccessible
+        print_section "Registry authentication and authorisation"
+        isAuthCorrect
+        print_section "Cluster operators"
+        verify_clusteroperators_status
+        print_section "Nodes"
+        verify_nodes_status
+        print_section "Machine configuration pools"
+        verify_mcp
+        print_section "Catalog sources"
+        verify_catsrc
+        print_section "Fusion software"
+        verify_fusion_health
+        print_section "Backup & Restore"
+        verify_br_health
+        print_section "Legacy SPP"
+        verify_spp_health
+        print_section "Data Classification"
+        verify_dcs_health
+        print_section "Scale cluster"
+        verify_scale_health
+        print_section "VMs migration"
+        verify_livemigratable_vms
+        print_section "Pods with imagepullbackoff across cluster"
+        verify_imagepullbackoff_pods
+        print_section "Nodes hardware status"
+        verify_nodes_hw
+        print_section "Fusion nodes maintenance"
+        verify_node_taints
+        print_section "Network checks"
+        verify_network_checks
+        print_section "Verify PID limit"
+        verify_pid_limit
+        print_section "Verify Presence of CSI Configmap"
+        verify_CSI_configmap_present
+        print_footer
+fi
+
+if [ "$1" == "postcheck" ]; then
```

Comment:
for post check also some basic check on API availability must be done else this pos check method itself will fail

Reply:
added basic api checks.